test/browser-providers.ts

```diff
@@ -0,0 +1,349 @@
+    os_version: '10'
+  },
+  'BS_WINDOWSPHONE' : {
+    base: 'BrowserStack',
+    device: 'Nokia Lumia 930',
+    os: 'winphone',
+    os_version: '8.1'
+  },
+  'BS_ANDROID5': {
+    base: 'BrowserStack',
+    device: 'Google Nexus 5',
+    os: 'android',
+    os_version: '5.0'
+  },
+  'BS_ANDROID4.4': {
+    base: 'BrowserStack',
+    device: 'HTC One M8',
+    os: 'android',
+    os_version: '4.4'
+  },
+  'BS_ANDROID4.3': {
+    base: 'BrowserStack',
+    device: 'Samsung Galaxy S4',
+    os: 'android',
+    os_version: '4.3'
+  },
+  'BS_ANDROID4.2': {
+    base: 'BrowserStack',
+    device: 'Google Nexus 4',
+    os: 'android',
+    os_version: '4.2'
+  },
+  'BS_ANDROID4.1': {
+    base: 'BrowserStack',
+    device: 'Google Nexus 7',
+    os: 'android',
+    os_version: '4.1'
+  }
+};
+
+const sauceAliases: AliasMap = {
+  'ALL': Object.keys(customLaunchers).filter(function(item) {
+    return customLaunchers[item].base == 'SauceLabs';
+  }),
+  'DESKTOP': ['SL_CHROME', 'SL_FIREFOX', 'SL_IE9', 'SL_IE10', 'SL_IE11', 'SL_EDGE', 'SL_SAFARI7',
+              'SL_SAFARI8', 'SL_SAFARI9'],
+  'MOBILE': ['SL_ANDROID4.1', 'SL_ANDROID4.2', 'SL_ANDROID4.3', 'SL_ANDROID4.4', 'SL_ANDROID5',
+             'SL_IOS7', 'SL_IOS8', 'SL_IOS9'],
+  'ANDROID': ['SL_ANDROID4.1', 'SL_ANDROID4.2', 'SL_ANDROID4.3', 'SL_ANDROID4.4', 'SL_ANDROID5'],
+  'IE': ['SL_IE9', 'SL_IE10', 'SL_IE11'],
+  'IOS': ['SL_IOS7', 'SL_IOS8', 'SL_IOS9'],
+  'SAFARI': ['SL_SAFARI7', 'SL_SAFARI8', 'SL_SAFARI9'],
+  'BETA': ['SL_CHROMEBETA', 'SL_FIREFOXBETA'],
+  'DEV': ['SL_CHROMEDEV', 'SL_FIREFOXDEV'],
+  'REQUIRED': buildConfiguration('unitTest', 'SL', true),
+  'OPTIONAL': buildConfiguration('unitTest', 'SL', false)
+};
+
+const browserstackAliases: AliasMap = {
+  'ALL': Object.keys(customLaunchers).filter(function(item) {
+    return customLaunchers[item].base == 'BrowserStack';
+  }),
+  'DESKTOP': ['BS_CHROME', 'BS_FIREFOX', 'BS_IE9', 'BS_IE10', 'BS_IE11', 'BS_EDGE', 'BS_SAFARI7',
+              'BS_SAFARI8', 'BS_SAFARI9'],
+  'MOBILE': ['BS_ANDROID4.3', 'BS_ANDROID4.4', 'BS_IOS7', 'BS_IOS8', 'BS_IOS9', 'BS_WINDOWSPHONE'],
+  'ANDROID': ['BS_ANDROID4.3', 'BS_ANDROID4.4'],
+  'IE': ['BS_IE9', 'BS_IE10', 'BS_IE11'],
+  'IOS': ['BS_IOS7', 'BS_IOS8', 'BS_IOS9'],
+  'SAFARI': ['BS_SAFARI7', 'BS_SAFARI8', 'BS_SAFARI9'],
+  'REQUIRED': buildConfiguration('unitTest', 'BS', true),
+  'OPTIONAL': buildConfiguration('unitTest', 'BS', false)
+};
+
+export const platformMap: { [name: string]: AliasMap } = {
+  'saucelabs': sauceAliases,
+  'browserstack': browserstackAliases,
+};
+
+
+/** Decode the token for Travis to use. */
+function decode(str: string): string {
+  return (str || '').split('').reverse().join('');
+}
+
+
+/** Setup the access keys */
+if (process.env.TRAVIS) {
+  process.env.SAUCE_ACCESS_KEY = decode(process.env.SAUCE_ACCESS_KEY);
+  process.env.BROWSER_STACK_ACCESS_KEY = decode(process.env.BROWSER_STACK_ACCESS_KEY);
+}
+
+/** Build a configuration. */
```

Comment:
This comment doesn't actually say anything (e.g., what a configuration consists of and what consumes it)